modules/openxr/openxr_api.cpp

```diff
@@ -1358,6 +1347,19 @@ void OpenXRAPI::register_extension_wrapper(OpenXRExtensionWrapper *p_extension_w
 	registered_extension_wrappers.push_back(p_extension_wrapper);
 }
 
+void OpenXRAPI::register_extension_metadata() {
+	for (OpenXRExtensionWrapper *extension_wrapper : registered_extension_wrappers) {
+		extension_wrapper->on_register_metadata();
```

Comment:
The contract of `on_register_metadata` is not clear to me. The methods seem to register everything provided by the extension unconditionally (even if the extension is not available), but given the way it is called here (also unconditionally) it seems that it then always registers the metadata?

Imho it would be best to keep the call here unconditional, but specify that `on_register_metadata` only registers metadata for available extensions, not for all of them. That also simplifies the overall interface of the wrapper, as there is no need to ever check anything outside of the wrapper (so the availability checks that seem unused can be safely removed).

Reply:
Ok so this is pretty important. The metadata is always needed. Remember, I do not have a PICO, yet I may still want to add PICO bindings to my action map for people who want to play my game on a PICO.

So the action map editor needs the meta data regardless of whether the computer the developer is on supports each device. So we always register the metadata even if the extension is not available.

We also use the metadata to process the actionmap. Originally I only had logic that would filter out paths for a disabled extension. The problem here is that any path with a typo in it would be seen as valid. Many XR runtimes actually crash if invalid paths are supplied to it.
Having a complete set of meta data allows us to validate paths and filter out invalid ones and alert the user properly, while checking valid paths against enabled extensions and filter them out if an extension is not enabled.

Reply:
You're right, of course. I was confused as I thought this PR is also supposed to move the filtering into the extension wrapper, but of course your change does not touch the old filters that are already in place. Sorry for my confusion!

I still think that might be a good idea for more flexibility (let each extension wrapper decide whether or not their actions should be passed on to OpenXR), but that is for a different PR.

Reply:
Well like I said, I kinda had it working like that originally and just found things became unstable because the items will be in the users action map. We loose the ability to know whether a path is invalid, or whether an extension filters out the path because it's not enabled. Though I guess that could be solved in other ways as well.

Reply:
Couldn't we directly pass an `available` or `enabled` parameter to `register_interaction_profile` and similar methods instead of an extension name?

If registration of metadata happens before the extensions are checked,  that parameter would be a boolean reference, else a plain boolean. All filtering logic can then happen in the extension wrapper.

Reply:
Indeed, we're registering everything before the extensions are checked, in fact the extensions are only checked in runtime (or we're in the VR editor).

Pushing pointers around is a problem as well, actually `get_requested_extensions` is a problem here as well as we take a boolean pointer here. This was fine in the Godot 3 plugin but poses a problem in 4 when implementing things in GDExtensions.
In the Vulkan drivers I've applied the same registration approach but we check if an extension was enabled by it's name. 
I may do the same here.

The way Khronos nowadays handles things, a lot of stuff is checked by name to make it as flexible as possible and as performance isn't really a consideration here.

That said, I get your point that we may want more flexibility in checking if a path is usable, we may at some point want to have the ability to set a callable function or something. But it all falls very much outside of the scope of this PR so as you say, food for thought for a follow up PR.